Document how to deploy Pulp on OpenShift (#149)
pulp/pulp-operator: docs/quickstart.md

# Getting started

> ## Insta demo

Reviewer: Is this "instant", or "installation"?

Author: I don't know, in every place I saw "insta" https://pulpproject.org/insta-demo/
@mikedep333 may know what it means

Reviewer 2: Or "Instance"?

Reviewer 3: "instant". A common marketing prefix in English.

> [A script](https://raw.githubusercontent.com/pulp/pulp-operator/master/insta-demo/pulp-insta-demo.sh)

Reviewer: Is it time to remove that script that is only there for "the time being"?

Is not considered production ready because pulp-operator is not yet, it hides every config option,
and upgrades are not considered. Only suitable as a quick way to evaluate Pulp for the time
being.

## OpenShift

> Currently pulp-operator is not on the OpenShift catalog, so as first step we need to create a catalog entry:

Reviewer: ```suggestion
Currently pulp-operator is not on the OpenShift catalog, so as a first step we need to create a catalog entry:
```

```yaml
# pulp-catalog-source.yaml
---
apiVersion: operators.coreos.com/v1alpha1
kind: CatalogSource
metadata:
 name: my-pulp-catalog
 namespace: openshift-marketplace
spec:
 sourceType: grpc
 image: quay.io/pulp/pulp-index:0.2.0

```
* Refer to [Getting started with the OpenShift CLI](https://docs.openshift.com/container-platform/4.7/cli_reference/openshift_cli/getting-started-cli.html)
```console
oc apply -f pulp-catalog-source.yaml
```

> Wait few seconds and refresh OCP page, after that you'll be able to see `my-pulp-catalog`

Reviewer: ```suggestion
Wait few seconds and refresh OCP page, after that you will be able to see `my-pulp-catalog`
```
maybe even use "should" instead of "will".

on the OperatorHub tab:
![OperatorHub tab](images/1.png "Pulp on OperatorHub tab")

> Click on `Pulp` and click on `Install` it:

Reviewer: ```suggestion
Click on `Pulp` and then `Install`:
```

![Installing pulp](images/2.png "Installing pulp operator")

![Installing pulp](images/3.png "Installing pulp operator")

Create a `Secret` with the `S3` credentials:
![S3 credentials Secret](images/4.png "S3 credentials Secret")

Click on `Pulp`:
![Click on Pulp](images/5.png "Click on Pulp")

> Select `S3` as storage type and `example-pulp-object-storage` on S3 storage secret:

Reviewer: That name is a free choice by the user, right?
It should be like "... and type the name of the storage you created before."

Author: yes, I was unsure on how to phrase it, because on the screenshot it is `example-pulp-object-storage`

Reviewer: Maybe: "... and type the name of the storage you created before, e.g. 'example-pulp-object-storage'."
